README.md

```diff
@@ -1 +1,72 @@
+- topologically sorting the dependency structure (to determine `include` order) is a burden placed upon the developer.
+
+## Solution
+
+The proposal is to extend the `using`/`import` syntax, by giving it a mode by which it can access files.
+
+Each file loaded in this manner would be evaluated in total isolation, stored globally in `PackageName.__toplevel__`, and a binding added to the current context.
+
+If a file has already been loaded then it would be looked up in the global reference rather than being re-evaluated, to avoid duplication issues.
+
+File identity is determined by filesystem location.
+
+If the specified file does not exist in the specified location, then an error is raised.
+
+Files are looked up relative to the filesystem location of the file in which the statement is written.
+
+As every file then imports its dependencies, then both of the major issues previously identified are resolved.
+
+## Syntax
+
+The suggested syntax is `from "../folder/file.jl" import myobj1, myobj2`, which would expect and require objects with names `:myobj1`, `:myobj2` to be defined inside `file.jl`. These objects could be modules, functions, etc.
+
+If all of `myobj1`, `myobj2`, etc. are modules, then `import` may be replaced with `using` to instead get access to all symbols exported by those modules.
+
+Two other alternate syntaxes for similar behaviour are `from ..folder.file import myobj1, myobj2` and `import "../folder/file.jl": myobj1, myobj2`. These are all roughly equivalent, so there are no strong feelings about which to use. (Those using `from` seem to read a little neater, but do introduce an extra keyword.)
+
+## Implementation
+
+The above is essentially syntactic sugar for:
+- Create `PackageName.__toplevel__` as a baremodule if it does not already exist.
+- If `PackageName.__toplevel__.Symbol("../folder/file.jl")` does not already exist:
+    - Create `PackageName.__toplevel__.Symbol("../folder/file.jl")` as a module.
+    - `include("../folder/file.jl")` into `PackageName.__toplevel__.Symbol("../folder/file.jl")`.
+- Evaluate one of the following expressions, according to the precise syntax used, where for readability we let `m` denote `PackageName.__toplevel__.Symbol("../folder/file.jl")`:
+    - `from "../folder/file.jl" import myobj1, myobj2`:  
+    `import m: myobj1, myobj2`
+    - `from "../folder/file.jl" import mymodule: myobj1, myobj2`:  
+    `import m.mymodule: myobj1, myobj2`
+    - `from "../folder/file.jl" import mymodule.myobj1, mymodule.myobj2`:  
+    `import m.mymodule.myobj1, m.mymodule.myobj2`
+    - `from "../folder/file.jl" using mymodule1, mymodule2`:  
+    `using m.mymodule1, m.mymodule2`
+    - `from "../folder/file.jl" using mymodule: myobj1, myobj2`:  
+    `using m.mymodule: myobj1, myobj2`
+
+## Alternate proposals
+
+One proposal was to use `import "../folder/file.jl"`, and to expect and require a module with name `:file` to be defined inside `"file.jl"`. However this has additional limitations:
```

Comment:
Let's use subsections so it's easier to read?

Reply:
That implies coming up with a title for each alternate proposal, though.